### PR TITLE
2242 "lean" docker compose, local key cloak, and minio

### PR DESCRIPTION
**Pull request by kbirk**
- merged some docker-compose overlays that were _always_ needed into the `docker-compose-common.yml` 
- added a `docker-compose-local-keycloak.yml` that supports the basic auth `service-user` when used in conjunction with `application-local-keycloak.properties` profile.
- added a singular `docker-compose-lean.yml` that stands a self contained `terarium` stack.
- Replaced `seaweedfs` with `minio`
- Fix some issues with presigned URL endpoints, now checks if the objcet exists for presignedGet urls.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,6 +154,27 @@ There should now be a `application-secrets.properties` file in the `packages/ser
...
+
+```shell
+docker compose --file containers/docker-compose-lean.yml pull
+docker compose --file containers/docker-compose-lean.yml up --detach --wait
+```
+
+This will deploy a local `terarium` backend along with all required services along with a localized `keycloak` instance to auth against. In order to hit the data service endpoints on port 3000 you will need to include the following basic auth credential:
+
+```
+'Authorization: Basic YXBpLXNlcnZpY2U6aWNvbiB2aW9sZXQgcmh5bWU='
```
Comment (dgauldie) [inline README.md:169]: we should have a fake credentials, or remove this.

Comment (dgauldie) [reply README.md:169]: or at least change the service account in keycloak with a different username and password for the local keycloak

**Review by dvince2 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (kbirk) [reply README.md:169]: It is, thats for the local one only

File: containers/docker-compose-lean.yml [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    restart: always
+    healthcheck:
+      test: [ "CMD", "redis-cli", "ping" ]
+      interval: "3s"
+      timeout: "3s"
+      retries: 30
+
+  minio:
+    image: quay.io/minio/minio:RELEASE.2023-11-15T20-43-25Z
+    command: [ "server", "/data", "--console-address", ":9090" ]
```
Comment (dgauldie) [inline containers/docker-compose-lean.yml:21]: can we add a container name for this.

File: containers/docker-compose-lean.yml [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    user: postgres
+    environment:
+      POSTGRES_USER: "postgres"
+      POSTGRES_PASSWORD: "postgres"
+      POSTGRES_DB: "postgres"
+    volumes:
+      - terarium-db:/var/lib/postgresql/data
+      - ./scripts/init.sql:/docker-entrypoint-initdb.d/init.sql
+
+  db-migration:
```
Comment (dgauldie) [inline containers/docker-compose-lean.yml:165]: do we need this anymore? or is it just for creating the databases?

File: containers/docker-compose-lean.yml [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+      SPRING_PROFILES_ACTIVE: "default, local-docker"
+      TERARIUM_DB_PASSWORD: "postgres"
+      TERARIUM_DB_USERNAME: "postgres"
+    depends_on:
+      postgres:
+        condition: service_healthy
+
+  keycloak:
+    image: ghcr.io/unchartedsoftware/keycloak:22.0.3.0
+    networks:
```
Comment (dgauldie) [inline containers/docker-compose-lean.yml:180]: can you add a container name

**Pull request merged**